persist/adapter.go

```diff
@@ -46,9 +46,15 @@ type Adapter interface {
 	// AddPolicy adds a policy rule to the storage.
 	// This is part of the Auto-Save feature.
 	AddPolicy(sec string, ptype string, rule []string) error
+	// AddPolicies adds policy rules to the storage.
+	// This is part of the Auto-Save feature.
+	AddPolicies(secs []string, ptypes []string, rules [][]string) []error
```

Comment:
It's completely unnecessary.

Reply:
Make sure these two functions are optional for an adapter.

Reply:
Okay,  got it will change this.

Reply:
There is a problem, if I remove these functions from the interface, I will be unable to use these functions for any adapter unless I make type assertions.
For example, ```e.adapter.(fileadapter).AddPolicies()``` this will work well, but if used like this ```e.adapter.AddPolicies()``` will throw an error. Golang doesn't support [dynamic type assertion](https://stackoverflow.com/questions/51504706/dynamic-type-assertion-golang) and we require this feature to get implemented on more than one adapters. 
There are two solutions either we add those functions in the interface or we go with a workaround like implementing a switch-case for each adapter which gets nasty.
So, what should we do?

Reply:
Any idea?

Reply:
As we know, ```e.adapter``` is an interface, if we remove those function signatures from there, it will not be able to access the functions at run-time.

Reply:
@nodece and @hsluoyz can you think of any way out of this, if we do not add these functions?

Reply:
Define a new interface called `AdapterEx`

Reply:
We should not add new methods to the `Adapter`. Just use like `AddPolicy` is good.

Reply:
> Define a new interface called `AdapterEx`. AdapterEx is a superset of Adapter.

Nice, working on this.

Reply:
> We should not add new methods to the `Adapter`. Just use like `AddPolicy` is good.

Yeah got it, thanks